src/view-canvas/app-view.ts

```diff
@@ -0,0 +1,104 @@
+import { Coords } from "../utility/coords";
+import { App } from "../app-states/app";
+import { Vector } from "../utility/vector";
+import { AnimationFrame } from "../utility/animation";
+
+export class Sprite {
+    constructor(private texture: CanvasImageSource) {}
+
+    draw(context: CanvasRenderingContext2D, options: {
+        position: Vector,
+        frame: AnimationFrame,
+        rotation: number
+    }) {
+        const translation = new Vector(
+            options.position.x,// + options.frame.w / 2,
+            options.position.y// + options.frame.h / 2
+        );
+
+        context.save();
+        context.translate(
+            translation.x, 
+            translation.y);
+        context.rotate((options.rotation + 90) * Math.PI/180.0);
+        context.translate(
+            -translation.x,
+            -translation.y);
+        context.drawImage(
+            this.texture, 
+            options.frame.x,
+            options.frame.y,
+            options.frame.w, 
+            options.frame.h, 
+            options.position.x - options.frame.w / 2,
+            options.position.y - options.frame.h / 2,
+            options.frame.w, 
+            options.frame.h);
+        context.restore();
+    }
+}
+
+export class AppViewCanvas {
+    private canvas2d;
+    private context2d: CanvasRenderingContext2D;
+    private texture: CanvasImageSource;
+    private ready = false;
+    private sprite: Sprite;
+
+    constructor(private app: App, canvasId: string) {
+        this.canvas2d = document.querySelector<HTMLCanvasElement>("#" + canvasId)!;
```

Comment:
Don't assume the element exists. Either passan existing one in the constructor, or (less preferably) creata the cancas here.